core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,591 @@
+            'android_report_info_schema_version':
+                base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'web_report_info': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'web_report_info_schema_version':
+                base_models.EXPORT_POLICY.NOT_APPLICABLE
+        })
+
+    @classmethod
+    def export_data(cls, user_id):
+        """Exports the data from AppFeedbackReportModel into dict format for
+        Takeout.
+
+        Args:
+            user_id: str. The ID of the user whose data should be exported;
+                this would be the ID of the user who has scrubbed the report.
+
+        Returns:
+            dict. Dictionary of the data from AppFeedbackReportModel.
+        """
+
+        user_data = dict()
+        report_models = cls.get_all().filter(
+            cls.scrubbed_by == user_id).fetch()
+
+        for report_model in report_models:
+            user_data[report_model.id] = {
+                'scrubbed_by': report_model.scrubbed_by,
+                'platform': report_model.platform,
+                'ticket_id': report_model.ticket_id,
+                'submitted_on': report_model.submitted_on.isoformat(),
+                'report_type': report_model.report_type,
+                'category': report_model.category,
+                'platform_version': report_model.platform_version
+            }
+        return user_data
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Model is exported as multiple instances per user since there
+        are multiple reports relevant to a user.
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.MULTIPLE_INSTANCES_PER_USER
+
+    @staticmethod
+    def get_lowest_supported_role():
+        """The lowest supported role for feedback reports will be moderator."""
+        return feconf.ROLE_ID_MODERATOR
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id):
+        """Check whether AppFeedbackReportModel exists for user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether a model is associated with the user.
+        """
+        return cls.query(
+            cls.scrubbed_by == user_id).get(keys_only=True) is not None
+
+
+class AppFeedbackReportTicketModel(base_models.BaseModel):
+    """Model for storing tickets created to triage feedback reports.
+
+    Instances of this model contain information about ticket and associated
+    reports.
+
+    The id of each model instance is created by combining the entity's
+    ticket_name hash, creation timestamp, and a random 16-character string.
+    """
+
+    # A name for the ticket given by the maintainer, limited to 100 characters.
+    ticket_name = datastore_services.StringProperty(required=True, indexed=True)
+    # The Github issue number that applies to this ticket.
+    github_issue_number = datastore_services.IntegerProperty(
```

Comment:
Do we also need to have a platform property in this model in order to understand which platform the GitHub issue corresponds to?

Reply:
I'm thinking that some issues may not be explicitly for one platform or the other. E.g. if there is an issue with a certain voiceover, this is a backend issue which affects both Web and Android, thought technically that could be categorized as Web based on repo location. The platform could also be decided once we start viewing the the individual reports in this ticket. That said, I think it's fine to include platform here as well since it does give more clarity and organization to what the ticket will address. WDYT?

Reply:
I think that if you can get an Android report which results in a web ticket being filed, you probably do need a field for which github repo the issue is being filed on.

Though I'm not sure you'd want to call it "platform" since that may differ from the platform that the report was submitted from, and using the same term for two different concepts might lead to confusion. Perhaps github_issue_tracker_repo_name or similar.

Reply:
Adding a field 'github_issue_repo_name'